contracts/contracts/strategies/qf-maci/MACIQF.sol

```diff
@@ -229,16 +247,23 @@ contract MACIQF is MACIQFBase, DomainObjs, Params {
 
         // Validate allowlist proof if provided GAS optimization
         // Don't check if the proof is empty
-        if (_proof.length != 0) {
-
-            uint256 watermark = allowlistVerifier.validateAllowlist(_proof);
-            if (watermark != uint256(uint160(_sender))) {
-                revert InvalidProof();
+        if (isAllowlisted) {
+            if (!_isAddressZero(address(allowlistVerifier))) {
+                bool verified = allowlistVerifier.validateUser(_proof, _sender);
+                if (!verified) {
+                    revert InvalidProof();
+                }
             }
             if (amount > maxContributionAllowlisted) {
                 revert ContributionAmountTooLarge();
             }
```

Comment:
If there's no `allowlistVerifier`, then you could provide `isAllowlisted = true` and be able to contribute up to the `maxContributionAllowlisted` amount. Probably would make more sense to revert in this case, e.g.:

```solidity
if (isAllowlisted) {
    if (_isAddressZero(address(allowlistVerifier))) {
        revert NoAllowlist();
    }
    bool verified = allowlistVerifier.validateUser(_proof, _sender);
    if (!verified) {
        revert InvalidProof();
    }
    ...
```

I guess in the below `else` statement, it's ok as is since if we don't have a `nonAllowlistVerifier` then we're ok with there not being any verification?